### PR TITLE
Task02

### PR DESCRIPTION
**Pull request by shagovAlexei**
edit Task02

### CODE REVIEW

File: tasks/task02/triangle.py [hunk 1]
```diff
@@ -2,49 +2,71 @@
...
-	side_a = int(parts[0])
-	side_b = parts[1]
-	angle = parts[2]
-	print(side_a, side_b, angle)
-	area = 0.5 * float(side_a) * float(side_b) * \
-            float(math.sin(math.radians(int(angle))))
-	return print("Area is: ", area)
+def check_positive(s):
+    try:
+        f = float(s)
```
Comment (ahodkov) [inline tasks/task02/triangle.py]: I don't found any information about float. And in the example used integer.

File: tasks/task02/triangle.py [hunk 2]
```diff
@@ -2,49 +2,71 @@
...
+        return False
+    except ValueError:
+        return False
+
+
+def calculate_are_by_base_and_height():
+    str = input("Enter your base and height (example: 5 6): ")
+    print_error = "Invalid input data! Enter two positive number"
+    parts = str.split(" ")
+    if len(parts) == 2:
```
Comment (ahodkov) [inline tasks/task02/triangle.py]: `if len(parts) == 2 and check_positive(parts[0]) and check_positive(parts[1]):`

File: tasks/task02/triangle.py [hunk 3]
```diff
@@ -2,49 +2,71 @@
...
+            print(print_error)
+    else:
+        print(print_error)
+
+
+def calculate_are_by_sides_and_angle():
+    str = input("Enter 2 sides and angle between them (example: 15 8 30): ")
+    print_error = "Invalid input data! Enter two positive number and angle >= 180"
+    parts = str.split(" ")
+    if len(parts) == 3:
```
Comment (ahodkov) [inline tasks/task02/triangle.py]: same

File: tasks/task02/triangle.py [hunk 4]
```diff
@@ -2,49 +2,71 @@
...
-		if playermenumain == "1":
-			Submenu_Base_Height()
-		elif playermenumain == "2":
-			Submenu_Sides_Angle()
-		elif playermenumain == "3":
-			print("Goodbye!")
-			break
+    menu_number = True
+    while menu_number:
+        print("""
```
Comment (ahodkov) [inline tasks/task02/triangle.py]: `menu='''
    Menu:
    1. Calculate triangle area by base and height
    2. Calculate triangle area by 2 sides and angle between them
    3. Exit
    '''`
`print(menu)`

File: tasks/task02/triangle.py [hunk 5]
```diff
@@ -2,49 +2,71 @@
...
+    """)
+        menu_number = input("Enter menu item number: ")
+        if menu_number == "1":
+            calculate_are_by_base_and_height()
+        elif menu_number == "2":
+            calculate_are_by_sides_and_angle()
+        elif menu_number == "3":
+            print("Goodbye!")
+            exit(0)
+        elif menu_number != "":
```
Comment (ahodkov) [inline tasks/task02/triangle.py]: `else:`

File: tasks/task02/triangle.py [hunk 6]
```diff
@@ -2,49 +2,71 @@
...
-
-		if playermenumain == "1":
-			Submenu_Base_Height()
-		elif playermenumain == "2":
-			Submenu_Sides_Angle()
-		elif playermenumain == "3":
-			print("Goodbye!")
-			break
+    menu_number = True
+    while menu_number:
```
Comment (ahodkov) [inline tasks/task02/triangle.py]: `menu_number != 3`

File: tasks/task02/triangle.py [hunk 7]
```diff
@@ -2,49 +2,71 @@
...
+        print("""
+    Menu:
+    1. Calculate triangle area by base and height
+    2. Calculate triangle area by 2 sides and angle between them
+    3. Exit
+    """)
+        menu_number = input("Enter menu item number: ")
```
Comment (ahodkov) [inline tasks/task02/triangle.py]: `menu_number = int(input("Enter menu item number: "))`

File: tasks/task02/triangle.py [hunk 8]
```diff
@@ -2,49 +2,71 @@
...
+        if menu_number == "1":
+            calculate_are_by_base_and_height()
+        elif menu_number == "2":
+            calculate_are_by_sides_and_angle()
+        elif menu_number == "3":
+            print("Goodbye!")
+            exit(0)
```
Comment (ahodkov) [inline tasks/task02/triangle.py]: if use `menu_number != 3` exit(0) doesn't need

**Review by ahodkov — CHANGES_REQUESTED** [review]

**Comment by shagovAlexei** [comment]
conflict

**Pull request closed without merging**